### PR TITLE
9 interview card data

### PR DESCRIPTION
**Pull request by jtmst**
This PR addresses issue [19](https://github.com/SuffolkLITLab/courtformsonline.org/issues/19) as well as issue [21](https://github.com/SuffolkLITLab/courtformsonline.org/issues/21)

### Changes:
- Splits TopicsCard into its own component file with updated logic for displaying interviews
- Adds fetchInterviewData file. 
- fetchInterviews logic.  Sorts interviews by topic.
- Simplify page.tsx.  Adds filter and sort logic for topic cards so that only topics with associated interviews are rendered (issue 21)
- Modify topic config data so all always_visible flags are set to false
- Modify server config file for easier addition of jurisdiction data later
- File change count inflated because of yarn cache refresh
- File organization changes, (config files in config dir)

image

### CODE REVIEW

**Comment by johnphamvan** [comment]
@nonprofittechy We acknowledge that the .yarn/cache makes these PRs very unwieldy. That's the current yarn best practice right now though. See https://github.com/yarnpkg/berry?tab=readme-ov-file#building-your-own-bundle. We recommend unchecking zip files from the file list.

image

**Comment by nonprofittechy** [comment]
No worries about the .zip files. I had some trouble running this branch locally but I'll come back with a more specific question.

**Comment by jtmst** [comment]
> No worries about the .zip files. I had some trouble running this branch locally but I'll come back with a more specific question.

Anything I can help out with getting it to run properly?

**Comment by nonprofittechy** [comment]
> > No worries about the .zip files. I had some trouble running this branch locally but I'll come back with a more specific question.
> 
> Anything I can help out with getting it to run properly?

So I did both a `yarn install` and then `yarn dev`, as well as `npm install` and `npm run start` but I just get a 404 error.

Maybe I'm having trouble understanding what the new routes are?

**Comment by jtmst** [comment]
> > > No worries about the .zip files. I had some trouble running this branch locally but I'll come back with a more specific question.
> > 
> > 
> > Anything I can help out with getting it to run properly?
> 
> So I did both a `yarn install` and then `yarn dev`, as well as `npm install` and `npm run start` but I just get a 404 error.
> 
> Maybe I'm having trouble understanding what the new routes are?

Routes should be unchanged and you should still see the homepage on localhost:3000/

**Comment by nonprofittechy** [comment]
This what I get. Note that I can run other node applications from the same install (and I could run this in the past) so it's not a firewall issue:

image

File: src/app/page.tsx [hunk 1]
```diff
@@ -79,29 +18,24 @@ export default function TopicsPage() {
...
-              .filter((topic: Topic) => topic.always_visible)
-              .map((topic: Topic) => (
-                <TopicCard key={topic.codes[0]} topic={topic} />
+              .sort((a, b) => b.priority - a.priority)
+              .filter(
+                (topic) =>
+                  topic.always_visible ||
+                  (interviewsByTopic[topic.name] &&
+                    interviewsByTopic[topic.name].length > 0)
+              )
```
Comment (nonprofittechy) [inline src/app/page.tsx:27]: LIST is a hierarchical code. 209A, which is FA-07-00-00-00, should go into the "Family" topic but it's getting sorted into Other

image

image

**Review by nonprofittechy — CHANGES_REQUESTED** [review]
This is looking good **except** we need to parse the LIST topics.

Edit: I suggest a function to see if a topic is contained in or is identical to a top level topic. The current site has some lower-level topics that we promoted to the "top" level, such as Appeals and Domestic Violence. So it's not just enough to strip off all of the numbers.

**Review by nonprofittechy — APPROVED** [review]

**Pull request merged**